lib/Index/IndexRecord.cpp

```diff
@@ -421,14 +423,15 @@ static void addModuleDependencies(ArrayRef<ImportedModule> imports,
           bool withoutUnitName = true;
           if (FU->getKind() == FileUnitKind::ClangModule) {
             auto clangModUnit = cast<ClangModuleUnit>(LFU);
-            if (!clangModUnit->isSystemModule() || indexSystemModules) {
-              withoutUnitName = false;
+            if ((!clangModUnit->isSystemModule() || indexSystemModules)) {
+              withoutUnitName = !indexClangModules;
               if (auto clangMod = clangModUnit->getUnderlyingClangModule()) {
                 moduleName = clangMod->getTopLevelModuleName();
```

Comment:
Also not sure if we need to fetch the proper module name here if the indexing is disabled.